.github/workflows/sanity.yml

```diff
@@ -127,3 +128,33 @@ jobs:
           make deploy
           kubectl get deployment -n ramen-system
           kubectl -n ramen-system wait --for=condition=Available --timeout=60s deploy/ramen-controller-manager
+
+  publish-image:
+    name: Publish built image
+    needs: [deploy-check, lint, golangci]
+    if: >
```

Comment:
this `if` check will get increased if we want to support more branches and tags. I think it's better to move this to a new file.

Reply:
There is a bit of [scripting tricks](https://github.com/backube/scribe/blob/2dc593bcd4c77e91c8195e9e6820487e53dfc3e5/.github/workflows/operator.yml#L277-L318) that can help, hoping that would ease the conditions in the future. If not we will move this to a separate file.